php-checks/src/test/resources/checks/UselessIncrementCheck.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+$a = 42;
+
+function pickNumber() {
+  $i = 0;
+  $j = 0;
+
+  $j = $j--; // Noncompliant {{Remove this decrement or correct the code not to waste it.}}
+  $i = $i++; // Noncompliant {{Remove this increment or correct the code not to waste it.}}
+//     ^^^^
+
+  $j += $j++; // Compliant
+  $i = ++$i;  // Compliant
+  $j = $i++;  // Compliant
+
+  if ($j < 42) {
+    return ($j)--; // Noncompliant
```

Comment:
also add a test case with upper / lower case difference

Reply:
Does not apply to variable names.